### PR TITLE
added authentiction with devise

### PR DESCRIPTION
**Pull request by admir23**
Hi, 

Please have a look at this? Hope this works for you.

Thanks
Admir

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (namangupta01) [inline app/models/user.rb]: Nice Work! Please also make it confirmable! So that user have to confirm before!

Comment (admir23) [reply app/models/user.rb]: @namangupta01 not sure what you mean about github configuration? can you help please?
I will make it confirmable too now.

File: db/migrate/20171015073149_devise_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+      ## Trackable
+      t.integer  :sign_in_count, default: 0, null: false
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.string   :current_sign_in_ip
+      t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
```
Comment (namangupta01) [inline db/migrate/20171015073149_devise_create_users.rb]: :confirmable won't work without this! So please also include this

Comment (namangupta01) [reply db/migrate/20171015073149_devise_create_users.rb]: so uncomment the confirmable section!

File: db/migrate/20171015073149_devise_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      t.datetime :current_sign_in_at
+      t.datetime :last_sign_in_at
+      t.string   :current_sign_in_ip
+      t.string   :last_sign_in_ip
+
+      ## Confirmable
+      t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (ViditChitkara) [inline db/migrate/20171015073149_devise_create_users.rb]: these lines should not be commented, since we may require them.

Comment (ViditChitkara) [reply db/migrate/20171015073149_devise_create_users.rb]: So you would have to rollback this migration uncomment this and again migrate.

Comment (ViditChitkara) [reply db/migrate/20171015073149_devise_create_users.rb]: Also read about smtp settings required to setup devise mailers. Without smtp settings in config/environment/developement.rb, confirmation email won't be sent.

**Comment by admir23** [comment]
ok, on it.  I've add this line to config/environments/developement.rb:

  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }

Is this ok?  I'm still pretty new with devise, so sorry if there is something wrong.

**Comment by admir23** [comment]
Or you want a smtp setting like this below:
onfig.action_mailer.default_url_options = { :host => 'localhost:3000' }

  config.action_mailer.delivery_method = :smtp
  config.action_mailer.perform_deliveries = true
  config.action_mailer.raise_delivery_errors = false
  config.action_mailer.default :charset => "utf-8"

  config.action_mailer.smtp_settings = {
    address: "smtp.gmail.com",
    port: 2525,
    domain: "gmail.com",
    authentication: "plain",
    enable_starttls_auto: true,
    user_name: "MY_EMAIL",
    password: "MY_PASS"
  }

**Comment by ViditChitkara** [comment]
Yep I was talking about this only!

**Comment by namangupta01** [comment]
@admir23 not all the lines
config.action_mailer.default_url_options = { :host => 'localhost:3000' }

  ActionMailer::Base.smtp_settings = {
  address:              'smtp.gmail.com',
  port:                 587,
  domain:               'gmail.com',
  user_name:            'myeamil',
  password:             'mypass',
  authentication:       :plain,
  enable_starttls_auto: true
}
as others line are already present so wehave to only add these lines only.

**Comment by namangupta01** [comment]
@admir23 It's look good now! Please squash the commits to one commit so that we can merge it!

**Comment by admir23** [comment]
ok, please have a look now.

**Comment by namangupta01** [comment]
@admir23 Please check your github configration, if it has wrong email configured then this issue will not be count in your credential. Your commit email is seems to be hello@admirkurtovic.info you can see this by typing git log on your system.
where i think your github email is hello@admirkurtovic.com. 
Just remember, both of these id should be same.

**Comment by admir23** [comment]
ok, thanks @namangupta01 
i've changed the my email to  hello@admirkurtovic.com in git config now... let me know if there is still a conflict.

**Comment by ViditChitkara** [comment]
@admir23 please squash your commits so that we can merge them

**Comment by namangupta01** [comment]
@admir23 Thanks :) 👍

**Comment by admir23** [comment]
thank you for you patience with me @ViditChitkara  and @namangupta01  :)

**Pull request merged**